config/config.go

```diff
@@ -52,6 +52,8 @@ type Config struct {
 	IdleTimeout             time.Duration             `yaml:"idle_timeout"`
 	DisableHTTPACValidation bool                      `yaml:"disable_http_ac_validation"`
 	DisableGRPCACDepsCheck  bool                      `yaml:"disable_grpc_ac_deps_check"`
+        ReadAfterWriteGuarantee bool                      `yaml:"read_after_write_guarantee"`
+        OverwriteCommited       bool                      `yaml:"overwrite_commited"`
```

Comment:
I think I would rather keep things simple and have only one behaviour, ie allow concurrent writes.

Reply:
Agree. Then ReadAfterWriteGuarantee configuration can be removed.

**What about OverwriteCommited?** Personally I don’t have a use case for replacing committed entries, because:

1. Either the cache is huge and contains so many artifacts from so many different builds, that it becomes too hard to know which entries are poisoned and should be overwritten. If I discovered that such a cache contained poisened entries, I would clear the whole cache anyway.
2. Or the cache is tiny and could be re-populated quickly, and then it would be OK to clear the whole cache.

Reply:
The only two reasons I can think why we would want to overwrite blobs are:
1) If the blob is somehow known to be corrupt, then we can make a client re-upload everything to replace the bad items.
2) For simplicity. We let the client upload the blob anyway, and saving it to disk is takes much less time than the upload so why not do it anyway, without bothering to check if the preexisting blob was corrupt?

Reply:
Good point about simplicity! And I have not noticed any obvious difference in performance. So the conclusion is to remove both configuration parameters and only keep the code for the cases ReadAfterWriteGuarantee=true and OverwriteCommited=true.

Reply:
Yes- let's not introduce configuration options, we just need to be careful to use atomic move/rename operations to replace existing files.